wdl-engine/src/backend/local.rs

```diff
@@ -0,0 +1,285 @@
+                    path = temp_dir.display()
+                )
+            })?;
+        }
+        fs::create_dir_all(&temp_dir).with_context(|| {
+            format!(
+                "failed to create directory `{path}`",
+                path = temp_dir.display()
+            )
+        })?;
+
+        Ok(Self {
+            work_dir: root.join("work"),
+            temp_dir,
+            command: root.join("command"),
+            stdout: root.join("stdout"),
+            stderr: root.join("stderr"),
+        })
+    }
+}
+
+impl TaskExecution for LocalTaskExecution {
+    fn map_path(&mut self, _: &Path) -> Option<PathBuf> {
+        // Local execution doesn't use guest paths
+        None
+    }
+
+    fn work_dir(&self) -> &Path {
+        &self.work_dir
+    }
+
+    fn temp_dir(&self) -> &Path {
+        &self.temp_dir
+    }
+
+    fn command(&self) -> &Path {
+        &self.command
+    }
+
+    fn stdout(&self) -> &Path {
+        &self.stdout
+    }
+
+    fn stderr(&self) -> &Path {
+        &self.stderr
+    }
+
+    fn constraints(
+        &self,
+        engine: &Engine,
+        requirements: &HashMap<String, Value>,
+        _: &HashMap<String, Value>,
+    ) -> Result<TaskExecutionConstraints> {
+        let num_cpus: f64 = engine.system().cpus().len() as f64;
+        let min_cpu = requirements
+            .get("cpu")
+            .map(|v| {
+                v.coerce(engine.types(), PrimitiveTypeKind::Float.into())
+                    .expect("type should coerce")
+                    .unwrap_float()
+            })
+            .unwrap_or(1.0);
+        if num_cpus < min_cpu {
+            bail!(
+                "task requires at least {min_cpu} CPU{s}, but only {num_cpus} CPU{s2} are \
+                 available",
+                s = if min_cpu == 1.0 { "" } else { "s" },
+                s2 = if num_cpus == 1.0 { "" } else { "s" }
+            );
+        }
+
+        let memory: i64 = engine
+            .system()
+            .total_memory()
+            .try_into()
+            .context("system has too much memory to describe as a WDL value")?;
+        let min_memory = requirements
+            .get("memory")
+            .map(|v| {
+                if let Some(v) = v.as_integer() {
+                    return Ok(v);
+                }
+
+                if let Some(s) = v.as_string() {
+                    return convert_unit_string(s)
+                        .and_then(|v| v.try_into().ok())
+                        .with_context(|| {
+                            format!("task specifies an invalid `memory` requirement `{s}`")
+                        });
+                }
+
+                unreachable!("value should be an integer or string");
+            })
+            .transpose()?
+            .unwrap_or(1);
+        if memory < min_memory {
+            bail!(
+                "task requires at least {min_memory} byte{s} of memory, but only {memory} \
+                 byte{s2} are available",
```

Comment:
This has to be "gigabytes" right? And same comment as above on the "is" vs. "are".

Reply:
It's converted to bytes before this and I didn't bother converting the bytes to a "min unit greater than 1" for the error message, though.

I'll see what I can do to improve the error message.

Reply:
Okay, I suppose this is tied in with the `unwrap_or()` above then—I was under the impression that represented 1 GiB, so if that represents 1 byte (and you're going to update to 2GiB), I get it.

It's _slightly_ better to report it as GiB if possible, but I don't think it's a show-stopper.